src/cli/sql.rs

```diff
@@ -102,6 +134,7 @@ pub async fn init(matches: &clap::ArgMatches) -> Result<(), Error> {
 						Statement::Use(stmt) => {
 							if let Some(namespace) = &stmt.ns {
 								ns = Some(namespace.clone());
+								db = None;
```

Comment:
This is intended to be a bugfix i.e. switching your namespace should automatically clear your currently selected database name.

Or is it reasonable to except that different namespaces will have equivalent table name(s)?

Reply:
If we decide to go this route, we will need to update the protocol and client as well to make sure the behaviour is consistent. Having said that, I think we shouldn't assume too much about the user's use cases. It's possible that the user may have namespaces with the similar databases. I think the current approach is more flexible.

Reply:
I've removed this change from this PR. While I think the current behavior is a foot-gun, especially in non-strict mode, we can discuss changing it in a future PR!